src/ipaperftest/core/main.py

```diff
@@ -424,7 +77,24 @@ def run(self, ctx):
         #       troubleshoot here
         for plugin in plugins:
             if plugin.__class__.__name__ == ctx.params['test']:
-                plugin.execute(ctx)
+                for result in plugin.execute(ctx):
```

Comment:
To catch all the random failures I think a try/except around this would do it. You'd have to create the Result yourself and I suppose include some of the traceback info.

Reply:
Thanks, that's a better way to handling this indeed. Added.